### PR TITLE
bugfix: fix deadcode reports by gometalinter

### LINKED ISSUE #2423: [linter issues] some deadcode are detected by tool gometalinter/deadcode
### Ⅰ. Issue Description

I used [deadcode](https://github.com/tsenart/deadcode) in [gometalinter](https://github.com/alecthomas/gometalinter) to check the code of PouchContainer project, then I found that there are some codes are unused:

```
ctrd/container.go:31:1:warning: defaultTrylockTimeout is unused (deadcode)
ctrd/utils.go:84:1:warning: generateID is unused (deadcode)
ctrd/utils.go:109:1:warning: ociImageToPouchImage is unused (deadcode)
daemon/containerio/hijack_conn.go:22:1:warning: createHijackConn is unused (deadcode)
daemon/daemon.go:56:1:warning: router is unused (deadcode)
daemon/daemon.go:120:1:warning: loadSymbolByName is unused (deadcode)
daemon/mgr/spec_linux.go:377:1:warning: isEmpty is unused (deadcode)
daemon/mgr/spec_linux.go:382:1:warning: isNone is unused (deadcode)
daemon/mgr/spec_linux.go:392:1:warning: isShareable is unused (deadcode)
daemon/mgr/spec_linux.go:403:1:warning: isPrivate is unused (deadcode)
main.go:262:1:warning: setLxcfsProcess is unused (deadcode)
pkg/utils/metrics/unit.go:7:1:warning: nanoseconds is unused (deadcode)
pkg/utils/metrics/unit.go:7:1:warning: bytes is unused (deadcode)
registry/auth.go:74:1:warning: genV1Endpoints is unused (deadcode)
storage/quota/grpquota.go:21:1:warning: grpQuotaType is unused (deadcode)
storage/quota/prjquota.go:21:1:warning: prjQuotaType is unused (deadcode)
test/util_api.go:34:1:warning: CreateBusyboxContainerOk is unused (deadcode)
test/util_api.go:68:1:warning: CreateBusybox125Container is unused (deadcode)
test/util_api.go:85:1:warning: StartContainerOk is unused (deadcode)
test/util_api.go:94:1:warning: StopContainerOk is unused (deadcode)
test/util_api.go:103:1:warning: CheckContainerStatus is unused (deadcode)
test/util_api.go:115:1:warning: CheckContainerRunning is unused (deadcode)
test/util_api.go:128:1:warning: DelContainerForceOk is unused (deadcode)
test/util_api.go:145:1:warning: PauseContainerOk is unused (deadcode)
test/util_api.go:154:1:warning: UnpauseContainerOk is unused (deadcode)
test/util_api.go:163:1:warning: DelContainerForceMultyTime is unused (deadcode)
test/util_api.go:191:1:warning: DelImageForceOk is unused (deadcode)
test/util_api.go:203:1:warning: CreateExecEchoOk is unused (deadcode)
test/util_api.go:226:1:warning: StartContainerExec is unused (deadcode)
test/util_api.go:239:1:warning: CreateVolumeOK is unused (deadcode)
test/util_api.go:256:1:warning: RemoveVolumeOK is unused (deadcode)
test/util_api.go:266:1:warning: DelNetworkOk is unused (deadcode)
test/util_api.go:275:1:warning: PullImage is unused (deadcode)
test/util_api.go:313:1:warning: GetMetric is unused (deadcode)
test/util_daemon.go:48:1:warning: StartDefaultDaemonDebug is unused (deadcode)
test/util_daemon.go:58:1:warning: StartDefaultDaemon is unused (deadcode)
test/util_daemon.go:68:1:warning: StartDaemonBareWithArgs is unused (deadcode)
test/util_daemon.go:75:1:warning: RestartDaemon is unused (deadcode)
test/util_daemon.go:81:1:warning: RunWithSpecifiedDaemon is unused (deadcode)
test/util_daemon.go:103:1:warning: CreateConfigFile is unused (deadcode)
test/utils.go:15:1:warning: helloworldImageOnlyRepoName is unused (deadcode)
test/utils.go:27:1:warning: testHubAddress is unused (deadcode)
test/utils.go:27:1:warning: testHubPasswd is unused (deadcode)
test/utils.go:27:1:warning: clientWrongCa is unused (deadcode)
test/utils.go:27:1:warning: testDaemonHTTPSAddr is unused (deadcode)
test/utils.go:27:1:warning: busyboxImage125ID is unused (deadcode)
test/utils.go:27:1:warning: testHubUser is unused (deadcode)
test/utils.go:62:1:warning: helpwantedForMissingCase is unused (deadcode)
test/utils.go:70:1:warning: SkipIfFalse is unused (deadcode)
test/utils.go:79:1:warning: IsTLSExist is unused (deadcode)
test/utils.go:102:1:warning: inspectFilter is unused (deadcode)
```

If these are indeed deadcode, I think we should try to eliminate them from this project. Since dead code will lead to increasing maintainace cost.

### Ⅱ. Describe what happened

deadcode exists somewhere in project.

### Ⅲ. Describe what you expected to happen
 no deadcode there.

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

none

### Ⅴ. Anything else we need to know?
none

### Ⅵ. Environment:

- pouch version (use `pouch version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ZYecho**
Signed-off-by: zhangyue <zy675793960@yeah.net>

### Ⅰ. Describe what this PR did
As the title desc.

### Ⅱ. Does this pull request fix one issue?

fix #2423 

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
None.

### Ⅳ. Describe how to verify it
None.

### Ⅴ. Special notes for reviews
The go tool's report about the deadcode have many errors, such as `PullImage` and `inspectFilter` func obviously being used in other code, maybe caused by lacking of `--tests` opt when use gometalinter.
I have double-check based on the reports.

updated:
still have errors even if use `gometalinter --tests --enable=deadcode ./...`
more deadcode not included in the report, such as `rootFSToAPIType` in `pouch/ctrd/utils.go`

### CODE REVIEW

File: test/util_api.go [hunk 1]
```diff
@@ -150,15 +141,6 @@ func PauseContainerOk(c *check.C, cname string) {
 	CheckRespStatus(c, resp, 204)
 }
 
-// UnpauseContainerOk unpauses the container and asserts success..
```
Comment (allencloud) [inline test/util_api.go]: For this part, I do not think removal of code is a good idea. Instead, I am afraid that we should try to take advantages of that in the test code, like:
```
	// unpause it
	resp, err = request.Post("/containers/" + cname + "/unpause")
	c.Assert(err, check.IsNil)
	CheckRespStatus(c, resp, 204)
```

Comment (allencloud) [reply test/util_api.go]: In addition, maybe in another pr, we could use `PauseContainerOk` to replace 
```
	resp, err := request.Post("/containers/" + cname + "/pause")
	c.Assert(err, check.IsNil)
	CheckRespStatus(c, resp, 204)
```

Comment (ZYecho) [reply test/util_api.go]: good catch, replace it in the next patch.

File: ctrd/utils.go [hunk 2]
```diff
@@ -3,24 +3,20 @@ package ctrd
...
 	"crypto/tls"
-	"encoding/json"
 	"net"
 	"net/http"
 	"time"
 
 	"github.com/alibaba/pouch/apis/types"
 	"github.com/alibaba/pouch/pkg/errtypes"
-	"github.com/alibaba/pouch/pkg/utils"
-
```
Comment (HusterWan) [inline ctrd/utils.go]: Keep the blank line, thanks

File: daemon/mgr/spec_linux.go [hunk 3]
```diff
@@ -373,45 +373,17 @@ func setupNamespaces(ctx context.Context, c *Container, specWrapper *SpecWrapper
 	return setupUtsNamespace(ctx, c, specWrapper)
 }
 
-// isEmpty indicates whether namespace mode is empty.
```
Comment (allencloud) [inline daemon/mgr/spec_linux.go]: cc @Ace-Tang to check this part.

File: main.go [hunk 4]
```diff
@@ -258,23 +257,6 @@ func initLog() {
 	logrus.SetFormatter(formatter)
 }
 
-// define lxcfs processe.
-func setLxcfsProcess(processes exec.Processes) exec.Processes {
```
Comment (allencloud) [inline main.go]: cc @CodeJuan to check lxcfs part.

File: ctrd/utils.go [hunk 5]
```diff
@@ -105,39 +89,6 @@ func rootFSToAPIType(rootFs *v1.RootFS) types.ImageInfoRootFS {
 	}
 }
 
-// ociImageToPouchImage transfer the image from OCI format to Pouch format.
```
Comment (allencloud) [inline ctrd/utils.go]: cc @fuweid to check image part.

Comment (fuweid) [reply ctrd/utils.go]: LGTM for image part.

**Comment by allencloud** [comment]
Thanks a lot for your work. @ZYecho This will definitely decrease the maintenance effort.

And please help continue a followup pr to take advantages of common code in the test code, like `PauseContainerOk `, `UnPauseContainerOk `.

**Pull request merged**